src/box/memtx_tree.cc

```diff
@@ -611,15 +634,38 @@ tree_iterator_start(struct iterator *iterator, struct tuple **ret)
 		 * last position in the tree, that's what we need.
 		 */
 		memtx_tree_iterator_prev(tree, &it->tree_iterator);
+		res = memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
+	}
+
+	/*
+	 * We need to use value semantics here, as the result's underlying data
```

Comment:
Is it possible to just set here `it->current = *res;` ?

Reply:
And set `*ret` here too. It seem that it would be shorter and will not require that comment.

Reply:
No, it’s not possible: see line 656 and corresponding branch.